architecture/2025-02-user-defined-resource-type-feature-spec.md

```diff
@@ -0,0 +1,1303 @@
+
+**Summary**
+
+When creating the resource type, the platform engineer can annotate the resource type to include developer documentation.
+
+**`postgreSQL-resource-type.bicep`**:
+
+```diff
+extension radius
+
+resource MyCompany.Data/postgreSQL 'System.Resources/resourceTypes@2023-10-01-preview' = {
+  name: 'MyCompany.Data/postgreSQL'
++   metadata: {
++    friendly-name: 'PostgreSQL'
++    description: '''
++        The MyCompany.Data/postgreSQL@v1alpha1 resource type
++        is a standard configuration relational database configured with 
++        corporate security settings enforced. 
++      
++        Example:
++        ...
++      
++        Owner:
++        author-platform-engineer@mycompany.com
++      
++        Change Log:
++        ...
++    '''
+  api: {
+    version: 'v1alpha1'
+    schema: {
+      properties: {
+        // Required input property
+        size: {
+          type: 'string'
+          enum: ['S', 'M', 'L', 'XL']
++          description: '''
++            The size of database to provision
++              - 'S': 0.5 vCPU, 2 GiB memory, 20 GiB storage
++              - 'M': 1. vCPU, 4 GiB memory, 40 GiB storage
++              - 'L': 2.0 vCPU, 8 GiB memory, 60 GiB storage
++              - 'XL': 4.0 vCPU, 16 GiB memory, 100 GiB storage
++            '''
+        }
+        // Optional input property
+        logging-verbosity: {
+          type: string
+          enum: ['TERSE', 'DEFAULT', 'VERBOSE']
++          description: '''
++            The logging level for the database
++              - 'TERSE': Not recommended; does not provide guidance on what to do about an error
++              - 'DEFAULT': Recommended level
++              - 'VERBOSE': Use only if you plan to actually look up the Postgres source code
++          '''
+        }
+        // Output property set by the recipe
+        connection-string: {
+          type: 'string'
+          readOnly: true
+          env-variable: POSTGRESQL_CONNECTION_STRING
++          description: 'Fully qualified string to connect to the resource'
+        }
+        // Output property set by the recipe
+        credentials: {
+          type: 'object'
+          readOnly: true
+          properties: {
+            username: {
+              type: 'string'
++              description: 'Username for the database'
+              env-variable: POSTGRESQL_USERNAME
+            }
+            password: {
+              type: 'string'
+              env-variable: POSTGRESQL_PASSWORD
++              description: 'Password for the database user'
+            }
+          }
+        }
+      // Specifying required properties
+      required: ['size']
+    }
+  }
+}
+```
+
+The description property on the resource type should be at least 2 KB, or approximately one page. Properties can be smaller.
```

Comment:
why do we need to specify this? is this something we're including in our docs?

Reply:
oh I see this is a note for the developer not to trim the response or something.